### PR TITLE
gpiozero SPIDevice - support for remote devices

### PR DESCRIPTION
**Pull request by mcarr823**
*Note: this PR is built on top of PR #116 and should be ignored until/unless that one is merged.

This PR furthers the implementation of the optional gpiozero dependency.
If gpiozero is detected, it will use gpiozero for SPI operations instead of using spidev directly.

Most interestingly, this allows for targeting remote devices which are attached to waveshare panels.
For example, running papertty on a desktop computer and sending updates to a raspberry pi on the same network.
This could facilitate some interesting use-cases, since a desktop PC is going to be significantly more powerful than a raspberry pi.

A quick example of remote execution and the setup necessary is below.

On the raspberry pi:
```bash
#Install gpiod
sudo apt install pigpio

#Remove the "-l" from the end of the ExecStart command
sudo vi /lib/systemd/system/pigpiod.service

#Reload the daemon, then start it
sudo systemctl daemon-reload
sudo systemctl start pigpiod.service

#Make it start automatically on reboot
sudo systemctl enable pigpiod.service
```

On the desktop:
```bash
#Install papertty like normal
#Then install gpiozero per the other PR
papertty_venv/bin/pip3 install gpiozero

#Then install pigpio
papertty_venv/bin/pip3 install pigpio

#Finally, start papertty REMOTELY
#It's the same command as you'd usually run for the panel, but with an environment variable for the IP address of the target device
sudo PIGPIO_ADDR=192.168.1.170 papertty_venv/bin/papertty --driver IT8951 terminal --size 12 --font myfont.ttf
```

### CODE REVIEW

File: papertty/drivers/driver_it8951.py [hunk 1]
```diff
@@ -188,21 +188,25 @@ def fixup_string(self, s):
...
+        #will sometimes occur during startup.
+        self.SPI = SpiDev()
+        self.SPI.setSpeed(2000000)
+        self.SPI.setMode(0b00)
+
         GPIO.setmode(GPIO.BCM)
         GPIO.setwarnings(False)
         GPIO.setup(self.RST_PIN, GPIO.OUT)
         GPIO.setup(self.CS_PIN, GPIO.OUT)
         GPIO.setup(self.BUSY_PIN, GPIO.IN)
```
Comment (joukos) [inline papertty/drivers/driver_it8951.py:202]: Now that these changes highlight it, it's a bit meh that this code is duplicated in `epd_init`, but that can be sorted out at a later time.

**Review by joukos — APPROVED** [review]
Thanks for this, using the display over network sounds like an interesting feature!

**Comment by joukos** [comment]
Oops, a wild conflict appeared.

File: papertty/drivers/driver_it8951.py (hunk 1 above)
Comment (mcarr823) [reply papertty/drivers/driver_it8951.py:203]: Yeah, I noticed that. The only difference is the DC pin, which could just be a flag on the epd_reset method.
I might add that to this PR.

I need to fix something else here anyway.
There's the conflict in IT8951 and also I missed the SPI read functionality in IT8951 (the base driver doesn't implement that, only IT8951 does).
I'll try to get those fixed up tomorrow.

Comment (mcarr823) [reply papertty/drivers/driver_it8951.py]: I've added the epd_init change in and fixed the IT8951 conflict after rebasing the changes on joukos:main
I've also re-tested on both the IT8951 and on the EPD3in7 to confirm that both work.

Gonna need to do some work on VNC/FB mode to take full advantage of this, but terminal mode at least is working quite well.
On CPU-limited devices (I tried with a Raspberry Pi Zero WH) the performance difference is massive, even with the added network latency.

**Pull request merged**